enterprise/server/remote_execution/runner/runner.go

```diff
@@ -246,42 +246,59 @@ func NewPool(cfg *config.RunnerPoolConfig) *Pool {
 // Add adds the given runner into the pool, evicting older runners if needed.
 // If an error is returned, the runner was not successfully added to the pool.
 func (p *Pool) Add(ctx context.Context, r *CommandRunner) error {
+	failureReason := ""
+	defer func() {
+		if failureReason != "" {
+			metrics.RunnerPoolFailedRecycleAttempts.With(prometheus.Labels{
+				metrics.RunnerPoolFailedRecycleReason: failureReason,
+			}).Inc()
+		}
+	}()
+
 	// TODO: once CommandContainer lifecycle methods are available, enforce that
 	// the runner's CommandContainer is paused, and return a
 	// FailedPreconditionError if not.
 
 	if r.state != ready {
+		failureReason = "unexpected_runner_state"
 		return status.InternalErrorf("unexpected runner state %d; this should never happen", r.state)
```

Comment:
this looks error-prone: it would be easy to add a new return statement and not update failureReason.

Could we add a layer of indirection (have Add call add()) and use the error String as our failure reason directly?

Reply:
Good call. I updated it to call `add()` but I made `add` return both an error and a short label for Prometheus (I'm not sure what the cardinality would be if we used the error message directly, since we form error strings from Docker errors and IDK if those include the container ID in them or other dynamic info). Since `add` returns `labeledError` and not `error` it's sort of a forcing function for people to remember to return a label whenever they return an error.